lang/cs/Org.Apache.REEF.Client/YARN/IJobSubmissionDirectoryProvider.cs

```diff
@@ -29,6 +29,6 @@ public interface IJobSubmissionDirectoryProvider
         /// Returns path to job submission directory in DFS.
         /// </summary>
         /// <returns></returns>
-        string GetJobSubmissionRemoteDirectory();
+        string GetJobSubmissionRemoteDirectory(string applicationId);
```

Comment:
Adding applicationId here makes it easier to debug. Not worth it to keep two methods.